make the order btw div and mul in adagrad update consistent (take 2)

jspark1105 opened the pull request:
Summary: D18703416 missed one place when we are making order mul and div in Adagrad consistent.

Test Plan: CI

Differential Revision: D18805426

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit a201e2bc:


* **1/1** broken upstream at merge base a54dc87e (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=a54dc87e8ebf7634d3fb2f32dd32b73c1a4d095f&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
  <details>
  <summary>
  You may want to rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  
* **0/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=a201e2bc1936e2afd9de4f2f057cb9b00f165640).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_linux_xenial_py3_clang5_asan_test](https://dr.pytorch.org/build-details.html?build_id=78663028)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jspark1105 committed:
make the order btw div and mul in adagrad update consistent (#230)

Summary:
Pull Request resolved: https://github.com/pytorch/FBGEMM/pull/230

Pull Request resolved: https://github.com/pytorch/pytorch/pull/30732

There was an inconsistency in the order of operation between scalar and SIMD code when we compute Adagrad. This diff make them consistent by computing the effective learning rate (lr / sqrt(momentum)) and then multiplying it with gradient.
This diff also uses fused multiply-add with a better accuracy and remove adagrad_avx.cc to use adagrad_avx2.cc instead.

Differential Revision: D18805426

fbshipit-source-id: 304e8dad3316518fa484d78a9811cd4dedab9618
